feat: Implement QuizTitle component

choiseona opened the pull request:
## 📂 작업 내용

closes #11 

- Default 컴포넌트 구현
- props로 title, currentRound, totalRound, remainingTime 받기

## 💡 자세한 설명

remainingTime이 10초 이하일 때 부터 깜빡거리는 타이머를 넣었습니다. 

## 📗 참고 자료 & 구현 결과 (선택)

![timer2](https://github.com/user-attachments/assets/2531ff9f-ccf3-4db3-9b1d-f415617332e6)

## 📢 리뷰 요구 사항 (선택)

title을 가운데에 위치시키기 위해 round 정보랑 타이머는 absolute로 두었습니다. 그런데 요소가 작아지면 absolute로 둔 round 정보와 타이머 그리고 title이 겹쳐지게 되는데 이걸 해결할 방법이 있을까요??

## 🚩 후속 작업 (선택)

## ✅ 셀프 체크리스트

- [x] PR 제목을 형식에 맞게 작성했나요?
- [x] 브랜치 전략에 맞는 브랜치에 PR을 올리고 있나요? (`main`이 아닙니다.)
- [x] 이슈는 close 했나요?
- [x] Reviewers, Labels를 등록했나요?
- [x] 작업 도중 문서 수정이 필요한 경우 잘 수정했나요?
- [x] 불필요한 코드는 제거했나요?

rhino-ty committed:
feat: Add timer images to /assets

choiseona committed:
feat: Add QuizTitle component

- Implement quizTitle component with round info and timer
- Add responsive styles for mobile and desktop
- Display dynamic timer with animation under 10 seconds

choiseona committed:
chore: Remove TimerHeader file

rhino-ty reviewed (approved):
수고하셨습니다!

겹치지 않게 하는 방법은 여러가지가 있는데

1. 타이틀 폰트 사이즈 더 작은 반응형에 의해서 작아지기
2. 타이틀과 문제 개수 칸을 2줄로 중앙 정렬화

뭐 이정도가 있겠네요! 

근데 애초에 **280px(Z폴드) ~ 320px(아이폰 SE)** 정도가 **화면 전체의 최소 width**로 둘 거라 감안하고 하셔야 합니다.
`min-w-[280px]` ~ `min-w-[360px]` 이런 설정을 해서 **레이아웃이 변형되지 않을 만큼**의 최소 width를 설정해 다양한 기기에 같은 레이아웃을 보여줘야하기 때문이죠!

rhino-ty commented on client/src/App.tsx:
```diff
@@ -19,31 +17,9 @@ const App = () => {
       </Button>
-      {/* UserInfoCard 예시 */}
-      <div className="space-y-8 p-4">
-        {/* 대기방 예시 */}
```
그러고보니 왜 내 `UserInfoCard 예시`는 지워버렸죠?! 네?! 합쳐주셔야죠!

choiseona commented:
> 수고하셨습니다!
> 
> 겹치지 않게 하는 방법은 여러가지가 있는데
> 
> 1. 타이틀 폰트 사이즈 더 작은 반응형에 의해서 작아지기
> 2. 타이틀과 문제 개수 칸을 2줄로 중앙 정렬화
> 
> 뭐 이정도가 있겠네요!
> 
> 근데 애초에 **280px(Z폴드) ~ 320px(아이폰 SE)** 정도가 **화면 전체의 최소 width**로 둘 거라 감안하고 하셔야 합니다. `min-w-[280px]` ~ `min-w-[360px]` 이런 설정을 해서 **레이아웃이 변형되지 않을 만큼**의 최소 width를 설정해 다양한 기기에 같은 레이아웃을 보여줘야하기 때문이죠!

해결해보려고했으나 잘 되지 않아서 요 부분은 조금 더 고민해 보고 리팩토링하겠습니다!!!

choiseona committed:
design: Make QuizTitle component width flexible

- Remove fixed max-width constraint
- Allow component to adapt to container width

rhino-ty commented:
> > 수고하셨습니다!
> > 겹치지 않게 하는 방법은 여러가지가 있는데
> > 
> > 1. 타이틀 폰트 사이즈 더 작은 반응형에 의해서 작아지기
> > 2. 타이틀과 문제 개수 칸을 2줄로 중앙 정렬화
> > 
> > 뭐 이정도가 있겠네요!
> > 근데 애초에 **280px(Z폴드) ~ 320px(아이폰 SE)** 정도가 **화면 전체의 최소 width**로 둘 거라 감안하고 하셔야 합니다. `min-w-[280px]` ~ `min-w-[360px]` 이런 설정을 해서 **레이아웃이 변형되지 않을 만큼**의 최소 width를 설정해 다양한 기기에 같은 레이아웃을 보여줘야하기 때문이죠!
> 
> 해결해보려고했으나 잘 되지 않아서 요 부분은 조금 더 고민해 보고 리팩토링하겠습니다!!!

나중에 레이아웃 작업하면서 추가로 수정할테니 너무 어려우면 넘겨도 됩니당

rhino-ty commented:
> > 수고하셨습니다!
> > 겹치지 않게 하는 방법은 여러가지가 있는데
> > 
> > 1. 타이틀 폰트 사이즈 더 작은 반응형에 의해서 작아지기
> > 2. 타이틀과 문제 개수 칸을 2줄로 중앙 정렬화
> > 
> > 뭐 이정도가 있겠네요!
> > 근데 애초에 **280px(Z폴드) ~ 320px(아이폰 SE)** 정도가 **화면 전체의 최소 width**로 둘 거라 감안하고 하셔야 합니다. `min-w-[280px]` ~ `min-w-[360px]` 이런 설정을 해서 **레이아웃이 변형되지 않을 만큼**의 최소 width를 설정해 다양한 기기에 같은 레이아웃을 보여줘야하기 때문이죠!
> 
> 해결해보려고했으나 잘 되지 않아서 요 부분은 조금 더 고민해 보고 리팩토링하겠습니다!!!

나중에 레이아웃 작업하면서 추가로 수정할테니 너무 어려우면 넘겨도 됩니당
